File: DSA/Aarush_590017879/Day_34/Q1_leetcode.cpp
#include <iostream>
#include <stack>
using namespace std;

class MyQueue {
private:
    stack<int> s1, s2;

    void transfer() {
        while (!s1.empty()) {
            s2.push(s1.top());
            s1.pop();
        }
    }

public:
    void push(int x) {
        s1.push(x);
    }

    int pop() {
        if (s2.empty())
            transfer();

        int val = s2.top();
        s2.pop();
        return val;
    }

    int peek() {
        if (s2.empty())
            transfer();

        return s2.top();
    }

    bool empty() {
        return s1.empty() && s2.empty();
    }
};

int main() {
    MyQueue q;

    q.push(1);
    q.push(2);
    q.push(3);

    cout << q.peek() << endl;
    cout << q.pop() << endl;
    cout << q.peek() << endl;
    cout << q.empty() << endl;

    return 0;
}

File: DSA/Navneetpant_590011569/Day_34/Q1_leetcode.java
class MyQueue { 
    Stack <Integer> queue = new Stack<>();

    public MyQueue() {
        
    }
    
    public void push(int x) {
       queue.push(x);
        
        
    }
    
    public int pop() {
        if (queue.isEmpty()) {
            return -1;
        }
        else {
            Stack <Integer> stack = new Stack<>();

            while (!queue.isEmpty()) {
                stack.push(queue.pop());
            }
            int temp = stack.pop();
            
            while (!stack.isEmpty()) {
                queue.push(stack.pop());
            }
            return temp;
        }
        
    }
    
    public int peek() {
         if (queue.isEmpty()) {
            return -1;
        }
        else {
            Stack <Integer> stack = new Stack<>();

            while (!queue.isEmpty()) {
                stack.push(queue.pop());
            }
            int temp = stack.peek();
            
            while (!stack.isEmpty()) {
                queue.push(stack.pop());
            }
            return temp;
           
        }

        
    }
    
    public boolean empty() {
        return queue.isEmpty();
        
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */